src/serde_util.rs

```diff
@@ -67,3 +67,62 @@ where
         (self.parse_fn)(v).map_err(E::custom)
     }
 }
+
+macro_rules! impl_tuple_visitor {
+    ($thing:ident, $len:expr) => {
+        /// Visitor type for de/ser a 32 byte sequence.
+        #[cfg(any(feature = "std", feature = "alloc"))]
+        pub struct $thing<F> {
+            expectation: &'static str,
+            parse_fn: F,
+        }
+
+        #[cfg(any(feature = "std", feature = "alloc"))]
+        impl<F, T, Err> $thing<F>
+        where
+            F: FnOnce(&[u8]) -> Result<T, Err>,
+            Err: fmt::Display,
+        {
+            pub fn new(expectation: &'static str, parse_fn: F) -> Self {
+                $thing {
+                    expectation,
+                    parse_fn,
+                }
+            }
+        }
+
+        #[cfg(any(feature = "std", feature = "alloc"))]
+        impl<'de, F, T, Err> de::Visitor<'de> for $thing<F>
+        where
+            F: FnOnce(&[u8]) -> Result<T, Err>,
+            Err: fmt::Display,
+        {
+            type Value = T;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str(self.expectation)
+            }
+
+            fn visit_seq<V>(self, mut seq: V) -> Result<Self::Value, V::Error>
+            where
+                V: de::SeqAccess<'de>,
+            {
+                let mut v = [0u8; $len];
+
+                for i in 0..$len {
```

Comment:
Why not `for byte_dst in &mut v`?

Or rather a better name than `v`. `bytes`?

Reply:
mutably iterate the vector, way better - thanks.